code/script.js

```diff
@@ -255,71 +294,125 @@ const checkQuestion = () => {
   // Compare the currentQuestion details with the secret person details in a different manner based on category (hair/eyes or accessories/others).
   // See if we should keep or remove people based on that
   // Then invoke filterCharacters
-  if (category === 'hair' || category === 'eyes') {
-
-  } else if (category === 'accessories' || category === 'other') {
 
+  if (category === "hair" || category === "eyes" || category === "gender") {
+    keep = value === secret[category];
+  } else if (category === "accessories" || category === "other") {
+    keep = secret[category].includes(value);
   }
-}
+  filterCharacters(keep);
+};
```

Comment:
I wrote this part a little different in my code, for the "hair" category i wrote for example:
"if (category === "hair") {
    keep = secretCharacter.hair === value;
  }"
It's cool to see how you made the code work for you!